x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sample/SampleIterator.java

```diff
@@ -84,13 +85,21 @@ public class SampleIterator implements Executable {
      */
     private long previousTotalPageSize = 0;
 
-    public SampleIterator(QueryClient client, List<SampleCriterion> criteria, int fetchSize, Limit limit, CircuitBreaker circuitBreaker) {
+    public SampleIterator(
+        QueryClient client,
+        List<SampleCriterion> criteria,
+        int fetchSize,
+        Limit limit,
+        CircuitBreaker circuitBreaker,
+        int maxSamplesPerKey
+    ) {
         this.client = client;
         this.criteria = criteria;
         this.maxCriteria = criteria.size();
         this.fetchSize = fetchSize;
         this.samples = new ArrayList<>();
         this.limit = limit;
+        this.maxSamplesPerKey = maxSamplesPerKey;
```

Comment:
After `circuitBreaker`, please. Keep the same order as the one of the constructor's arguments.